libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -2234,7 +2234,6 @@ MQTTStatus_t MQTT_GetIncomingPacketTypeAndLength( TransportRecv_t readFunc,
     }
     else if( ( status != MQTTBadParameter ) && ( bytesReceived == 0 ) )
     {
-        LogError( ( "No data was received from the transport." ) );
```

Comment:
Removed this error log since it's not an error for data not to be available from the transport. I chose not to make it a debug log since this function can be called continuously in a loop by MQTT_ProcessLoop and MQTT_ReceiveLoop, and could result in lots of printed messages if the socket receive timeout is small.

Reply:
Assuming that debug is the most verbose log level and not default, do you think the information "MQTT_GetIncomingPacketTypeAndLength failed to receive any data from transport" can still be useful while trying to trace the behavior?

Reply:
I'm not entirely sure. Given the fact that this function is run in a loop, if debug logs are enabled I'd think the clutter caused by repeatedly printing the same log message might make it harder to trace behavior instead of easier. Since the return code in this situation is already set to the unique `MQTTNoDataAvailable`, I didn't see the need to also log it. I can add it back as a debug log if you think it adds value

Reply:
It can be useful in some circumstances- for example, I enable debug logs and see this message printed multiple times before success. This probably tells me to look for if MQTT loop timeout or transport receive timeout is too short. 

I do understand your concern about too many messages - but that happening on the most verbose level should be okay.